### DISCUSSION TITLE
Working on API fork: How to track workflow ending

### DISCUSSION THREAD

**Discussion by rawmen0x** [comfyanonymous/ComfyUI#773] [Q&A]
Hi,

First, thanks for the great project for everyone working on. 

I'm testing an API version, but notice some difficult on tracking the end of an workflow.

I'm tracking currently on **send_sync** function on server.py, but none of the signals seems to be working.

data['value'] == ['max'] will finish before other branches on the workflow

event == 'status' or 'executed' does not work too.

Is there any signal a full workflow finished?

Cheers.

**Comment by comfyanonymous** [comment]
https://github.com/comfyanonymous/ComfyUI/blob/master/script_examples/websockets_api_example.py

Maybe this will help.

**Reply by rawmen0x** [reply]
Thank you kindly!

**Reply by rawmen0x** [reply]
Makes sense and it's now functional and working.
If anyone find yourself in the same issue, the event tracking is:
Signals in send_sync:
if event == 'executing':
  if data['node'] is None and data['prompt_id'] == prompt_id:
      #execution is complete

Thank you.
